feature-libs/asm/components/customer-emulation/customer-emulation.component.ts

```diff
@@ -25,12 +48,62 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
     );
     this.isCustomerEmulationSessionInProgress$ =
       this.asmComponentService.isCustomerEmulationSessionInProgress();
+
+    this.subscription.add(
+      this.cartId.valueChanges.subscribe((response: string) => {
+        this.cartIdExists = response.trim().length > 0;
+      })
+    );
```

Comment:
Form field validation should use angular reactive form built-in validation system instead of creating an observable stream for that purpose.